linkerd/proxy/client-policy/src/lib.rs

```diff
@@ -469,12 +494,17 @@ pub mod proto {
                         Some(section)
                     };
 
+                    let port = u16::try_from(port)
+                        .ok()
+                        .and_then(|p| NonZeroU16::try_from(p).ok());
```

Comment:
nit: `NonZeroU16::try_from(p).ok()` is equivalent to [`NonZeroU16::new`](https://doc.rust-lang.org/stable/std/num/struct.NonZeroU16.html#method.new), which returns an `Option`:
```suggestion
                        .and_then(NonZeroU16::new)
```